integration/util/registry.go

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) 2023 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+package util
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+const (
+	configDefaultMode = 0666
+
+	systemctl = "systemctl"
+)
+
+type Resource struct {
+	url string
+
+	method string
+	body   string
+
+	user string
+	pass string
+
+	delete bool
+}
+
+type BootstrapConfiguration struct {
+	LogFile             string   `json:"logFile"`
+	PostBootstrapFile   string   `json:"postBootstrapFile"`
+	PostBootstrapScript []string `json:"postBootstrapScript"`
+	CaCert              string   `json:"caCert"`
+	Address             string   `json:"address"`
+	TenantID            string   `json:"tenantId"`
+	DeviceID            string   `json:"deviceId"`
+	AuthID              string   `json:"authId"`
+	Password            string   `json:"password"`
+}
+
+type ConnectorConfiguration struct {
+	CaCert   string `json:"caCert"`
+	LogFile  string `json:"logFile"`
+	Address  string `json:"address"`
+	TenantID string `json:"tenantId"`
+	DeviceID string `json:"deviceId"`
+	AuthID   string `json:"authId"`
+	Password string `json:"password"`
+}
+
+// CreateDeviceResources creates device resources.
+func CreateDeviceResources(newDeviceId, tenantID, policyID, password, registryAPI,
+	registryAPIUsername, registryAPIPassword string, cfg *TestConfiguration) []*Resource {
+
+	devicePath := tenantID + "/" + newDeviceId
+	return []*Resource{
+		&Resource{url: registryAPI + "/devices/" + devicePath, method: http.MethodPost,
+			body: `{"authorities":["auto-provisioning-enabled"]}`,
+			user: registryAPIUsername, pass: registryAPIPassword, delete: true},
+		&Resource{url: registryAPI + "/credentials/" + devicePath, method: http.MethodPut,
+			body: getCredentialsBody(strings.ReplaceAll(newDeviceId, ":", "_"), password),
+			user: registryAPIUsername, pass: registryAPIPassword, delete: false},
+		&Resource{url: GetThingURL(cfg.DigitalTwinAPIAddress, newDeviceId), method: http.MethodPut,
+			body: fmt.Sprintf(`{"policyId": "%s"}`, policyID),
+			user: cfg.DigitalTwinAPIUsername, pass: cfg.DigitalTwinAPIPassword, delete: true},
+	}
+}
```

Comment:
I think it would be more readable to initialise the structures like this:

```suggestion
func CreateDeviceResources(newDeviceId, tenantID, policyID, password, registryAPI,
	registryAPIUsername, registryAPIPassword string, cfg *TestConfiguration) []*Resource {
	devicePath := tenantID + "/" + newDeviceId
	return []*Resource{
		&Resource{
			url:    registryAPI + "/devices/" + devicePath,
			method: http.MethodPost,
			body:   `{"authorities":["auto-provisioning-enabled"]}`,
			user:   registryAPIUsername,
			pass:   registryAPIPassword,
			delete: true},
		&Resource{
			url:    registryAPI + "/credentials/" + devicePath,
			method: http.MethodPut,
			body:   getCredentialsBody(strings.ReplaceAll(newDeviceId, ":", "_"), password),
			user:   registryAPIUsername,
			pass:   registryAPIPassword,
		},
		&Resource{
			url:    GetThingURL(cfg.DigitalTwinAPIAddress, newDeviceId),
			method: http.MethodPut,
			body:   fmt.Sprintf(`{"policyId": "%s"}`, policyID),
			user:   cfg.DigitalTwinAPIUsername,
			pass:   cfg.DigitalTwinAPIPassword,
			delete: true},
	}
}
```